src/include/os_fhandle.i

```diff
@@ -6,6 +6,68 @@
  * See the file LICENSE for redistribution information.
  */
 
+/*
+ * __wt_stat_read_io_histogram --
+ *	Add an operations execution time to the disk read histogram.
+ */
+static inline void
+__wt_stat_read_io_histogram(WT_SESSION_IMPL *session, uint64_t msecs)
+{
+	/*
+	 * Ignore any operation that takes less than 10ms to execute. This
+	 * floor value keeps us from having an excessively large smallest
+	 * execution time bucket. The 10ms value and the other histogram buckets
+	 * were chosen based on expectations of disk response times and testing.
+	 */
+	if (msecs < 10)
+		return;
+
+	if (msecs < 50) {
```

Comment:
I am also heavily in favor of this suggestion. I think you could create macros for them to incorporate the statistic field name via `##name` in a manner similar to the atomic functions in `gcc.h`. However where I disagree with @agorrod is that this PR has been open for 5+ months, it is worth spending a day more to get the function structure right and be done with it rather than postpone in another ticket.